src/Hedgehog/Plutus/Tx.hs

```diff
@@ -104,7 +112,125 @@ data ScriptPurpose
   In the future, hps should support custom balancers.
 -}
 balanceTx :: Model.Mock -> Tx 'Unbalanced -> Maybe (Tx 'Balanced)
-balanceTx = _
+balanceTx mock tx = do
+  let balanceingKeys = Map.keys $ Model.mockUsers mock
+  -- It probably doesn't make sense to balance a transaction
+  -- as every user at once in most cases
+  -- so we should probably make this an argument to balance
+  let valueOut = getValueOut tx
```

Comment:
You should probably use `toLedgerTx` and  [`evaluateTransactionBalance`](https://github.com/input-output-hk/cardano-ledger/blob/e0947b0c2e99ef929d20593be03e199560392f42/eras/shelley/impl/src/Cardano/Ledger/Shelley/API/Wallet.hs#L496) to calculate balances

Reply:
That took way longer than I expected lots of digging through https://github.com/input-output-hk/cardano-ledger ,https://github.com/input-output-hk/cardano-base and https://github.com/input-output-hk/plutus.